src/basic/env-util.c

```diff
@@ -207,6 +211,74 @@ static int env_append(char **r, char ***k, char **a) {
         return 0;
 }
 
+int strv_env_load_with_callback(char ***l, char **fnames, void (*invalid_callback)(const char *file, char ***p, void *userdata), void *userdata) {
+        char **i, **r = *l;
```

Comment:
Looks like _cleanup_strv_free_ should be used for *r. This would avoid the need for all the explicit cleanup below.